core/config/config.go

```diff
@@ -430,6 +430,23 @@ func (c *Config) Init() error {
 	return c.parseChains()
 }
 
+// Return the schema to use for accessing the node.
+func (c *Config) Schema() (s string) {
+	s = "https"
+	if c != nil && c.DisableHttps {
+		s = "http"
+	}
+	return
+}
+
+// Convenience method to check if HTTPS is enabled.
+func (c *Config) IsHttpsEnabled() bool {
```

Comment:
lets drop this method - there are only 3 callsites, easy enough to just use the value.